eth/gasprice/gasprice.go

```diff
@@ -147,24 +148,30 @@ func (gpo *Oracle) SuggestPrice(ctx context.Context) (*big.Int, error) {
 }
 
 type getBlockPricesResult struct {
-	prices []*big.Int
-	err    error
+	price *big.Int
+	err   error
 }
 
 // getLowestPrice calculates the lowest transaction gas price in a given block
 // and sends it to the result channel. If the block is empty, price is nil.
-func (gpo *Oracle) getBlockPrices(ctx context.Context, blockNum uint64, ch chan getBlockPricesResult) {
+func (gpo *Oracle) getBlockPrices(ctx context.Context, signer types.Signer, blockNum uint64, ch chan getBlockPricesResult) {
 	block, err := gpo.backend.BlockByNumber(ctx, rpc.BlockNumber(blockNum))
 	if block == nil {
 		ch <- getBlockPricesResult{nil, err}
 		return
 	}
 	txs := block.Transactions()
-	prices := make([]*big.Int, len(txs))
-	for i, tx := range txs {
-		prices[i] = tx.GasPrice()
+	var minPrice *big.Int
+	for _, tx := range txs {
+		sender, err := types.Sender(signer, tx)
+		if err != nil || sender == block.Coinbase() {
+			continue
+		}
+		if minPrice == nil || tx.GasPrice().Cmp(minPrice) < 0 {
+			minPrice = tx.GasPrice()
```

Comment:
`tx.GasPrice` creates a copy internally. Calling it twice here will alloc twice. It would be cleaner to do:

```go
if curPrice := tx.GasPrice(); minPrice == nil || curPrice.Cmp(minPrice) < 0 {
 	minPrice = curPrice
}
```